core/translation_service.py

```diff
@@ -0,0 +1,104 @@
+import hjson
+
+from core.decorators import instance
+from core.event_service import EventService
+from core.logger import Logger
+from core.setting_service import SettingService
+from core.setting_types import TextSettingType
+from core.tyrbot import Tyrbot
+from core.util import Util
+
+
+@instance()
+class TranslationService:
+    strings = {}
+    modules = {}
+    language = "en_US"
+    last_file = None
+    lang_codes=["en_US", "de_DE"]
+
+    def __init__(self):
+        self.logger = Logger(__name__)
+
+    def inject(self, registry):
+        self.setting_service: SettingService = registry.get_instance("setting_service")
+        self.event_service: EventService = registry.get_instance("event_service")
+        self.util: Util = registry.get_instance("util")
+        self.bot: Tyrbot = registry.get_instance("bot")
+
+    def pre_start(self):
+        self.language = self.setting_service.get_value("language") or self.language
+        self.event_service.register_event_type("reload_translation")
+
+    def start(self):
+        self.setting_service.register("language", "en_US", "Language of the Bot",
+                                      TextSettingType(self.lang_codes), "core.system")
+        self.registerTranslation("global", self.read_translations_from_file("core/global.msg"))
+
+    def registerTranslation(self, category, translations):
+        for k in translations:
+            if self.strings.get(category):
+                self.strings[category][k] = translations[k].get(self.language) or translations[k].get("en_US")
+            else:
+                self.strings[category] = {k: translations[k].get(self.language) or translations[k].get("en_US") + "\""}
+        if self.last_file is not None:
+            if self.modules.get(category) is None:
+                self.modules[category] = []
+            self.modules[category].append(self.last_file)
+
+        self.last_file = None
+
+    def reload_translation(self, lang):
+        self.bot.send_private_channel_message("<orange>My language got changed. reloading translations...<end>")
+        self.bot.send_org_message("<orange>My language got changed. reloading translations...<end>")
```

Comment:
you will need to add fire_outgoing_event=false to both of these calls otherwise those messages will get displayed twice

Reply:
see: https://github.com/Budabot/Tyrbot/blob/master/modules/core/system/system_controller.py#L60-L61